hathor/simulator/simulator.py

```diff
@@ -141,38 +139,41 @@ def stop(self) -> None:
         self._started = False
         self._patches_rc_decrement()
 
-    def create_peer(
-        self,
-        network: Optional[str] = None,
-        peer_id: Optional[PeerId] = None,
-        enable_sync_v1: bool = True,
-        enable_sync_v2: bool = True,
-        soft_voided_tx_ids: Optional[Set[bytes]] = None,
-        full_verification: bool = True,
-        event_ws_factory: Optional[EventWebsocketFactory] = None
-    ) -> HathorManager:
+    def get_default_builder(self) -> Builder:
+        """
+        Returns a builder with default configuration, for convenience when using create_peer() or create_artifacts()
+        """
+        return Builder() \
+            .set_network(self._network) \
+            .set_soft_voided_tx_ids(set()) \
+            .enable_full_verification() \
+            .enable_sync_v1() \
+            .enable_sync_v2() \
+            .use_memory()
+
+    def create_peer(self, builder: Builder) -> HathorManager:
+        """
+        Returns a manager from a builder, after configuring it for simulator use.
+        You may get a builder from get_default_builder() for convenience.
+        """
+        artifacts = self.create_artifacts(builder)
+        return artifacts.manager
+
+    def create_artifacts(self, builder: Builder) -> BuildArtifacts:
+        """
+        Returns build artifacts from a builder, after configuring it for simulator use.
+        You may get a builder from get_default_builder() for convenience.
+        """
         assert self._started, 'Simulator is not started.'
-        assert peer_id is not None  # XXX: temporary, for checking that tests are using the peer_id
```

Comment:
@msbrogli do you know why we had this? I removed it because the behavior is unchanged: if the user of this method doesn't `set_peer_id()` on the builder he's passing, an exception will be raised.

If we were to allow a default `peer_id`, we could `set_peer_id(PeerId())` on `get_default_builder()`, and then the `builder` argument on `create_peer()` and `create_artifacts()` could be optional, simplifying its use for basic use cases:

```python
def create_artifacts(self, builder: Optional[Builder] = None) -> BuildArtifacts:
    builder = builder or self.get_default_builder()
    ...
```

Reply:
I guess it was an old change to prevent using default peer ids. I don't feel we need it anymore.